addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -171,6 +171,30 @@ const FontFamilyPickerUserValueWidget = SelectUserValueWidget.extend({
                                 return;
                             }
                         }
+
+                        let isValidFamily = true;
+
+                        try {
+                            const result = await fetch("https://fonts.googleapis.com/css?family=" + m[1]);
+                            // We shouldn't rely on CORS to know if a family is valid. Keeping this in case of a change from Google.
+                            // Google fonts server returns a 400 status code if family is not valid.
+                            if (!result.ok) {
+                                isValidFamily = false;
+                            }
+                        } catch (error) {
+                            // At the moment, a CORS error is thrown if a family name does not exist.
+                            if (error instanceof TypeError) {
+                                isValidFamily = false;
+                            } else {
+                                throw error;
+                            }
+                        }
```

Comment:
```suggestion
                        let isValidFamily = false;

                        try {
                            const result = await fetch("https://fonts.googleapis.com/css?family=" + m[1], {method: 'HEAD'});

                            // Google fonts server returns a 400 status code if family is not valid.
                            if (result.ok) {
                                isValidFamily = true;
                            }
                        } catch (error) {
                            console.error(error);
                        }
```

not tested but should be enough.

HEAD avoid to download content if not needed ;)